arch-snapshot/arch.md

```diff
@@ -48,7 +48,57 @@ only as detailed as to explain how the actors control the artifacts. The goal is
 to give the reader an overview of the terms introduced by Golem without any
 details and establish a glossary to ensure consistency within the document.
 
-### Selling on golem platform
+### Selling on Golem platform
+
+This chapter explains the process of advertising resources on the Golem Network and selling
+them to Requestors. It provides a high-level overview of how the Provider Agent application
+should operate and interact with the Yagna Daemon.
+
+The following steps are needed:
+1. Describe Resources using property language to create [Offer](#offer)
+2. Publish Offer on market
+3. Listen on incoming Proposal events and negotiate [Agreement](#agreement) with the most promising Requestor
+4. Wait until Requestor will demand [Activity](#activity) creation by listening to activity events
+5. Allocate promised Resources according to [Agreement](#agreement)
+6. Send [DebitNotes](#debit-note) to notify Requestor with update cost
+7. Terminate Agreement or wait for Agreement termination event sent by Requestor
+8. Send [Invoice](#invoice) to summarize the cost of the Agreement
+9. Listen on Payment API events for Invoice settlement and payment confirmation 
+
+#### 1. Describe Resources using property language to create [Offer](#offer)
+
+The Golem design was created to support the sale of any type of computing resource.To achieve this,
```

Comment:
I think we need an explanation of what computational resources are and what constraints we have decided to put on what is being sold on Golem.